### PR TITLE
🔏 Requires valid access tokens for API calls

### LINKED ISSUE #153: Setup API token authentication
Setup authentication for ~~eventual~~ existing API routes
```[tasklist]
### Tasks
- [x] Create Chowda API
- [x] Create Chowda API permissions
- [x] Create Mario app (Machine to Machine)
- [x] Assign Chowda API permissions to Mario
- [x] Create Chowda JWT validation function
- [x] Add JWT validation dependency to API routes
- [ ] Check `permissions` for authorization
- [x] Add tests
- [ ] https://github.com/WGBH-MLA/mario/issues/18
```

### PR DESCRIPTION
**Pull request by afred**
# Verified Access Token
* Adds a verified access token as a dependency to API endpoints, including;
  * `POST /api/events/`
  * `POST /api/sony_ci/sync`
* This same dependency can now be added to any API endpoint.
* The verified access token dependency is resolved by:
  1. Fetching the unverified access token from the HTTP request header.
  2. Verifying the access key by decoding it with the signing key.
     a. In the production and development  environments, the signing key is the
        public half of a public/private key pair available with from our Auth0 account.
     b. In test environment a fake signing key is used for both encoding and decoding.
* Adds tests for testing the authentication and authorization dependencies of the API
  endpoints.
  * The testing pattern involves a factory for generating a fake access tokens with
    whatever permissions the test is concerned with testing.
* Refactors the FastAPI dependencies for API endpoints for  better testability.

## Misc
* Removes unused launch config for tests. Use VSCode pytest settings instead.
* Changes param name from allow_redirects to follow_redirects to fix deprecation warning
* Adds git-guardian cache dir to git ignore
* Adds constants to config used for verifying access token and specifying the authorization `audience`.
* Renames `admin_user` dependency to  to `get_admin_user` for consistency.
* Renames `get_user` to `get_oauth_user` for specificity -- i.e. not a User model from the Chowda db.
* Adds pyjwt for dealing with Javascript Web Tokens (jwt).
* Removes test for api calls to create batches - not needed.
* Removes `.env` files from revision, but leaves a sample template: `.env.sample`.
* Prefixes all env vars and constants having to do with Auth0 with `"AUTH0_"` for consistency.
* Sets test db used in GitHub CI workflow to use name `'chowda'` and default port `5432`.

Closes #153

### CODE REVIEW

File: chowda/routers/sony_ci.py [hunk 1]
```diff
@@ -12,7 +14,9 @@ class SyncResponse(BaseModel):
 
 
 @sony_ci.post('/sync', tags=['sync'])
-async def sony_ci_sync() -> SyncResponse:
+async def sony_ci_sync(
+    token: Annotated[OAuthAccessToken, Depends(verified_access_token)]
```
Comment (mrharpo) [inline chowda/routers/sony_ci.py:18]: Need to check for `sync:sonyci` permissions here.

**Review by mrharpo — CHANGES_REQUESTED** [review]
👍 🔏 Just some minor fixes, plus I've added a few more tests.

**Review by afred — COMMENTED** [review]
👍 from me. I the rest is good, then :shipit:

**Review by mrharpo — APPROVED** [review]
👍 👍

**Pull request merged**